### ISSUE TITLE
inconsistent results in `subdivide` for rects divided six or seven times

### ISSUE THREAD

**Issue by benalbrecht** [thi-ng/geom#70]
Expected:

```clojure
(count (geom/subdivide (rect/rect 1) {:cols 5, :rows 1})) => 5
(count (geom/subdivide (rect/rect 1) {:cols 6, :rows 1})) => 6
(count (geom/subdivide (rect/rect 1) {:cols 7, :rows 1})) => 7
(count (geom/subdivide (rect/rect 1) {:cols 8, :rows 1})) => 8
```

Actual:
```clojure
(count (geom/subdivide (rect/rect 1) {:cols 5, :rows 1})) => 5
(count (geom/subdivide (rect/rect 1) {:cols 6, :rows 1})) => 7
(count (geom/subdivide (rect/rect 1) {:cols 7, :rows 1})) => 8
(count (geom/subdivide (rect/rect 1) {:cols 8, :rows 1})) => 8
```

The root cause seems to be [in these two lines](https://github.com/thi-ng/geom/blob/feature/no-org/src/thi/ng/geom/rect.cljc#L272-L273) with the following results:
```clojure
((juxt count last) (range 0.0 1.0 (/ 1.0 5))) => [5 0.8]
((juxt count last) (range 0.0 1.0 (/ 1.0 6))) => [7 0.9999999999999999]
((juxt count last) (range 0.0 1.0 (/ 1.0 7))) => [8 0.9999999999999998]
((juxt count last) (range 0.0 1.0 (/ 1.0 8))) => [8 0.875]
```

**Comment by postspectacular** [comment]
Ah yes, the joys of floating point math... 😭 It seems I forgot to update the `subdivide` impl for rectangles as I'd already done for the 3D counterpart (`AABB` shape type), i.e. we should use `norm-range` from the [thi.ng/math](https://github.com/thi-ng/math/blob/master/src/core.org#intervals) package (the last function in that linked code block), which is more precise than the Clojure built-in `range` function, since it doesn't accumulate the floating point error caused by repeated addition of the range step value:

To compare, `Rect2` still uses `range`:

https://github.com/thi-ng/geom/blob/feature/no-org/src/thi/ng/geom/rect.cljc#L272

But AABB already uses `norm-range`:

https://github.com/thi-ng/geom/blob/feature/no-org/src/thi/ng/geom/aabb.cljc#L258

(note: the `butlast` calls are needed since `norm-range` uses a closed interval (i.e. always includes 1.0 at the end), whereas `range` is semi-open...

```clj
(pprint (g/subdivide (a/aabb 1) {:cols 7, :rows 1 :slices 1}))
({:p [0.0 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.14285714285714285 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.2857142857142857 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.42857142857142855 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.5714285714285714 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.7142857142857143 0.0 0.0], :size [0.14285714285714285 1.0 1.0]}
 {:p [0.8571428571428571 0.0 0.0], :size [0.14285714285714285 1.0 1.0]})
```

It's an easy change, so would appreciate a PR, but no prob if you don't feel like it! If you're up for it, please make sure it's against the `feature/no-org` branch...

**Comment by benalbrecht** [comment]
PR created, thanks for the directions :)

**Comment by postspectacular** [comment]
thank you! let's keep this issue open until I've made the release...

**Comment by benalbrecht** [comment]
released in 1.0.0-RC4 :+1:
